bip-v1-packages.mediawiki

```diff
@@ -0,0 +1,314 @@
+
+# '''Total fees and weight.''' The 'total_fee' and 'total_weight' fields must accurately represent the sum total of all transactions' fees and weights as defined in BIP141, respectively.
+
+Not all of the child's parents must be present; the child transaction may also spend confirmed
+inputs. However, if the child has confirmed parents, they must not be in the package.
+
+While a child-with-unconfirmed-parents package is perhaps most relevant when the child has a higher
+feerate than its parents, this property is not required to construct a valid package.
+
+===Clarifications===
+
+'''Q: Under what circumstances should a sender announce a child-with-unconfirmed-parents package?'''
+
+A child-with-unconfirmed-parents package for a transaction should be announced when it meets the
+peer's fee filter but one or more of its parents don't; a "inv(MSG_PCKG1)" instead of "inv(WTX)"
+should be sent for the child. Each of the parents which meet the peer's fee filter should still be
+announced normally.
+
+'''Q: What if a new block arrives in between messages?'''
+
+A child-with-unconfirmed-parents package is defined for a transaction based on the current chain
+state. As such, a new block extending the tip may decrease the number of transactions in the package
+(i.e. if any of the transaction's parents were included in the block). In a reorg, the number of
+transactions in the package may decrease or increase (i.e. if any of the transaction's parents were
+included in a block in the previous chain but not the new one).
+
+If the new block arrives before the "getdata" or "pckginfo1", nothing needs to change.
+
+[[File:./bip-v1-packages/newblock_benign.png|500px]]
+
+''Diagram: Benign "new block" scenarios.''
+
+If the new block arrives before "getpckgtxns" or before "pckgtxns", the receiver may need to
+re-request package information if the block contained a transaction in the package. If the block
+doesn't contain any transactions in the package, whether it extends the previous tip or causes a
+reorg, nothing needs to change.
+
+[[File:./bip-v1-packages/newblock_rerequest.png|1000px]]
+
+''Diagram: "New block" scenarios in which additional messages must be sent.''
+
+'''Q: Can "getpckgtxns" and "pckgtxns" messages contain only one transaction?'''
+
+Yes.
+
+==Compatibility==
+
+Older clients remain fully compatible and interoperable after this change. Clients implementing this
+protocol will only attempt to send and request packages if agreed upon during the version handshake.
+
+===Package Erlay===
+
+A client using BIP330 reconciliation-based transaction relay (Erlay) is able to use package relay
+without interference. In fact, a package of transactions may be announced using both Erlay and
+package relay.  After reconciliation, if the initiator would have announced a transaction by wtxid
+but also has package information for it, they may send "inv(MSG_PCKG)" instead of "inv(WTX)".
+
+[[File:./bip-v1-packages/package_erlay.png|900px]]
```

Comment:
In the erlay picture, are some of the arrows at the bottom of the image needing to be reversed?

Reply:
Thanks! fixed